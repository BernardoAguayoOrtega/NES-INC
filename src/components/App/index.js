// import react
import React from 'react';

//create and export app component
const App = () => {
  return (
    <>
      <h1>I&apos;m the app component xd</h1>
    </>
  );
};

export default App;
