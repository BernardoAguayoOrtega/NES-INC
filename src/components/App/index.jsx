// import react
import React from 'react';
// import hash router
import { HashRouter as Router, Switch, Route } from 'react-router-dom';
// import Home page
import Home from '../../pages/Home/index.jsx';
// import team page
import Team from '../../pages/Team/index.jsx';
// import article page
import Article from '../../pages/Article/index.jsx';
//import about page
import About from '../../pages/About/index.jsx'
// import layout component
import Layout from '../Layout/index.jsx';
// import page error
import Error from '../Error/index.jsx';

//create and export app component
const App = () => {
  return (
    <Router basename="/">
      <Layout>
        <Switch>
          <Route exact path="/">
            <Home />
          </Route>
          <Route path="/team">
            <Team />
          </Route>
          <Route path="/about">
            <About />
          </Route>
          <Route exact path="/article/:name">
            <Article/>
          </Route>
          <Route>
            <Error
              message="ERROR 404"
              subtitle="Ups looks like a problem"
            />
          </Route>
        </Switch>
      </Layout>
    </Router>
  );
};

export default App;
