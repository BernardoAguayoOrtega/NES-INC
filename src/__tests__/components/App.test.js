// import react
import React from 'react';
// import shallow
import { shallow } from 'enzyme';
// import app component
import App from '../../components/App';
// import toJson
//import toJson from 'enzyme-to-json';

describe('<App />', () => {
  const wrapper = shallow(<App />);

  it('is render the app?', () => {
    expect(wrapper.exists()).toBe(true)
  });

  it('are there home page?', () => {
    expect(wrapper.find('Home').exists()).toBe(true)
  });
});
