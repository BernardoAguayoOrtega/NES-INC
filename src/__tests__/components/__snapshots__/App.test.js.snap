// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snap shot 1`] = `
<HashRouter
  basename="/"
>
  <Layout>
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <Home />
      </Route>
      <Route
        path="/team"
      >
        <Team />
      </Route>
      <Route
        path="/about"
      >
        <About />
      </Route>
      <Route
        exact={true}
        path="/article/:name"
      >
        <Article />
      </Route>
      <Route
        path="/Privacy_Policy"
      >
        <Policy />
      </Route>
      <Route>
        <Error
          message="ERROR 404"
          subtitle="This page does not exist, keep browsing, we have much more news :)"
        />
      </Route>
    </Switch>
  </Layout>
</HashRouter>
`;
